Automattic/wp-calypso
# Hosting: Add static file 404 handling UI

#### Proposed Changes

This creates a Web Server Settings card that includes
the ability to set both PHP Version and how the web server
handles requests for nonexistent static assets. Previously,
PHP Version had its own card, but it is now part of the
Web Server Settings card.

Old PHP Version card
![php-version-card](https://user-images.githubusercontent.com/530877/173627227-b9a882de-03fc-423b-8363-25cf92982f58.png)

New Web Server Settings card
![web-server-settings](https://user-images.githubusercontent.com/530877/173627357-b9240ed8-8f8b-448b-bf19-6af0ab631051.png)

#### Testing Instructions

1. Pick a WoA test site.
2. In a browser, navigate to "https://your-test-site/nonexistent.js" and note that you see a rich, WordPress-generated 404 page.
3. Open the Calypso hosting page for the test site and scroll to the Web Server Settings card.
4. Change the "Handling Requests for Nonexistent Assets" to "Send a lightweight File-Not-Found Page"
5. Refresh the browser tab for nonexistent.js and observe you get a simple nginx 404 page.
6. Try the other options, refresh nonexistent.js, and observe you get the response you expect. The current Default is to use a WordPress-generated 404 page, but we may change it sometime in the future.
